plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/decoders/Decoder.java

```diff
@@ -16,9 +16,10 @@
 import io.trino.spi.block.BlockBuilder;
 import org.opensearch.search.SearchHit;
 
+import java.util.List;
 import java.util.function.Supplier;
 
 public interface Decoder
 {
-    void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output);
+    void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output, List<String> dereferenceName);
```

Comment:
`dereferenceName` is it a name or rather a path?
To consider `dereferencePath`.